crates/cheatcodes/src/inspector.rs

```diff
@@ -1348,13 +1351,17 @@ impl Cheatcodes {
             // Record frame paused gas.
             self.paused_frame_gas.push(interpreter.gas);
         }
+    }
+
+    #[cold]
+    fn meter_gas_end(&mut self, interpreter: &mut Interpreter) {
+        fn will_exit(ir: InstructionResult) -> bool {
```

Comment:
makes sense! was looking into this as without https://github.com/foundry-rs/foundry/pull/8743/commits/55627ed32e95421e626c6f10c0cb5ddf7b951018  test 8383 output gas was increased by 100 for any new call added in pause/resume block. Ideally `test_repro!(8383);` should be changed to smth like

```Solidity
// https://github.com/foundry-rs/foundry/issues/8383
test_repro!(8383, false, None, |res| {
    let mut res = res.remove("default/repros/Issue8383.t.sol:Issue8383Test").unwrap();
    let test = res.test_results.remove("testP256VerifyOutOfBounds()").unwrap();
    assert_eq!(test.status, TestStatus::Success);
    assert!(test.short_result("").contains("(gas: 3103)"));
});
```

as the initial fix accounted only for test to pass (no panic) but not for constant gas too.
Would be OK to add a commit with test changed here, or should I follow up with a different PR?

Reply:
@DaniPopes I made PR https://github.com/foundry-rs/foundry/pull/8744 with addition above and fix for #4370 , happy to include it here or with different PR after this one is merged